### ISSUE TITLE
Can connect same calendar multiple times

### ISSUE THREAD

**Issue by Pradumn27** [calcom/cal.com#10455]
Found a bug? Please fill out the sections below. 👍

### Issue Summary

We're able to connect the same calendar multiple times, have tested with Google and Outlook Calendars but I guess it's present for all calendars.

https://github.com/calcom/cal.com/assets/47187878/6d79b74b-46d7-445a-876a-371fa10a1d94

<img width="1440" alt="Screenshot 2023-07-28 at 9 10 29 PM" src="https://github.com/calcom/cal.com/assets/47187878/d4d02081-95df-41ab-8626-6c9f26329a40">

### Steps to Reproduce

1. Install any calendar app
2. Connect an email
3. Now connect the same email again and again

### Technical Details

1. Upon checking the code we're basically not checking for already existing calendars in the callback functions.
2. The caveat is that it will increase the time to connect a calendar.

**Comment by Akhilbisht798** [comment]
Yes, i can confirm this. I can add same calendar multiple time in my app. Also i would like to work on this issue.

**Comment by ajayjha1** [comment]
I am working on this issue but installing calendars needs authentication and authentication requires HTTPS. Is there any way to authenticate locally?
